pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -456,61 +461,170 @@ func (b *propBuf) FlushLockedWithRaftGroup(
 			continue
 		}
 
+		// Determine which encoding we're going to use for the Entry.Data slice.
+		var entryEncoding raftlog.EntryEncoding
+		switch {
+		case p.command.ReplicatedEvalResult.AddSSTable != nil:
+			entryEncoding = raftlog.EntryEncodingSideloaded
+		case p.command.ReplicatedEvalResult.ChangeReplicas != nil:
+			// Flush any previously batched (non-conf change) proposals to
+			// preserve the correct ordering or proposals. This isn't
+			// strictly necessary for anything - after all, latching prevents
+			// reordering where it matters, but it's not worth changing now.
+			// Later proposals will start a new batch.
+			if err := proposeBatch(raftGroup, b.p.getReplicaID(), ents); err != nil {
+				firstErr = err
+				continue
+			}
```

Comment:
reflection: Another reason to decouple this from the encoding `switch-case` is: this `proposeBatch()` then wouldn't need to move to a different place, the diff would be cleaner and easier to review.